src/Application/Digital/Commands/ShareEntry.cs

```diff
@@ -0,0 +1,178 @@
+                    var childAllowOperations = GenerateAllowOperations(request, childEntry.IsDirectory);
+                    await GrantPermission(childEntry, user, childAllowOperations, request.ExpiryDate, cancellationToken);
+                }
+            }
+
+            var result = await _context.EntryPermissions.FirstOrDefaultAsync(x =>
+                    x.EntryId == request.EntryId
+                    && x.EmployeeId == request.UserId
+                , cancellationToken);
+            
+            await _context.SaveChangesAsync(cancellationToken);
+            return _mapper.Map<EntryPermissionDto>(result);
+        }
+
+        private async Task GrantPermission(
+            Entry entry,
+            User user,
+            string allowOperations,
+            DateTime expiryDate,
+            CancellationToken cancellationToken)
+        {
+            var existedPermission = await _context.EntryPermissions.FirstOrDefaultAsync(x =>
+                    x.EntryId == entry.Id
+                    && x.EmployeeId == user.Id
+                , cancellationToken);
+            
+            if (existedPermission is null)
+            {
+                var entryPermission = new EntryPermission
+                {
+                    EmployeeId = user.Id,
+                    EntryId = entry.Id,
+                    AllowedOperations = allowOperations,
+                    ExpiryDateTime = LocalDateTime.FromDateTime(expiryDate),
+                    Employee = user,
+                    Entry = entry,
+                };
+                await _context.EntryPermissions.AddAsync(entryPermission, cancellationToken);
+            }
+            else
+            {
+                existedPermission.AllowedOperations = allowOperations;
+                existedPermission.ExpiryDateTime = LocalDateTime.FromDateTime(expiryDate);
+                _context.EntryPermissions.Update(existedPermission);
+            }
+            await _context.SaveChangesAsync(cancellationToken);
+        }
+
+        private static string GenerateAllowOperations(Command request, bool isDirectory)
+        {
+            if (request is { CanView: false, CanDownload: false, CanUpload: false, CanChangePermission: false })
+            {
+                return string.Empty;
+            }
+            
+            var allowOperations = new CommaDelimitedStringCollection();
+
+            if (request.CanView)
+            {
+                allowOperations.Add(EntryOperation.View.ToString());
+            }
+            
+            if (request is { CanView: true, CanUpload: true } && !isDirectory)
+            {
+                allowOperations.Add(EntryOperation.Upload.ToString());
+            }
+            
+            if (request is { CanView: true, CanDownload: true } && !isDirectory)
+            {
+                allowOperations.Add(EntryOperation.Download.ToString());
+            }
+            
+            if (request.CanChangePermission)
```

Comment:
I think this too require a CanView permission to grant

Reply:
Sure

Reply:
Fixed at https://github.com/ProFile-org/ProFile-BE/pull/336/commits/996b088078782ad3ddd0ff05eeed54cfa4c31a63